app/infrastructure/scrapers/lms_scraper.py

```
@@ -19,376 +19,182 @@
 class LmsScraper:
-    def __init__(self, client: LmsClient) -> None:
+    def __init__(self, client: LmsClient, parser: LmsParser) -> None:
         self.client = client
+        self.parser = parser
 
     async def _fetch(
         self,
         method: HTTPMethod,
         url: str,
+        data: Optional[dict] = None,
+        check: Optional[bool] = False,
```

Ruff (0.12.2)

Boolean-typed positional argument in function definition

(FBT001)